### DISCUSSION TITLE
Error of XMLHttpRequest

### DISCUSSION THREAD

**Discussion by SearaChen** [jspsych/jsPsych#606] [Q&A] [answered]
When I launch an old html built with jspsych with chrome I am running into the following error, even though the script has worked fine before 
“jspsych.js:2198 Access to XMLHttpRequest at 'file:///Users/Seara/Desktop/prototype_experiment/mp3s/CelineDion1.mp3' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https.
load_audio_file_webaudio @ jspsych.js:2198”

I have never encountered this error before, and the bug seem to be coming from jsPsych. Do you have any idea as to why is this happening?

**Comment by jodeleeuw** [comment, marked as answer]
This is a security feature of web browsers. If you run the experiment on a web server the problem will be solved. You can also [disable the security](https://stackoverflow.com/questions/3102819/disable-same-origin-policy-in-chrome) for local testing.

**Comment by haribharadwaj** [comment]
We ran into this issue when running from a web server (Apache), but realized it was because WebAudio needs to read audio files that we we hosted on a different server (AWS S3 bucket). But once we added CORS headers that that S3 can serve along with the audio files, browsers seemed OK to let the WebAudio scripts run with cross-origin media. Just wanted to mention that here as an example of this issue in a server context although OP's issue seems to because of running locally.

**Reply by jackransomlovell** [reply]
Can you point me to the example you mentioned? Or your code :)? I am trying to host a large set of stimuli on an s3 bucket, and preload them into my experiment but keep getting a CORS error, even though my CORS configuration allows access to the server I'm running my experiment on. Thanks!

**Reply by haribharadwaj** [reply]
@jack-rl-ovell, did you add CORS headers **on S3** as in https://docs.aws.amazon.com/AmazonS3/latest/user-guide/add-cors-configuration.html? We just used the example file provided here: https://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html.  Also, clearing browser cache and reloading experiment might be needed.
